mllib/src/main/scala/org/apache/spark/ml/feature/HashingTF.scala

```diff
@@ -42,14 +42,23 @@ import org.apache.spark.util.VersionUtils.majorMinorVersion
  * otherwise the features will not be mapped evenly to the columns.
  */
 @Since("1.2.0")
-class HashingTF @Since("1.4.0") (@Since("1.4.0") override val uid: String)
+class HashingTF @Since("3.0.0") private[ml] (
+    @Since("1.4.0") override val uid: String,
+    @Since("3.1.0") val hashFuncVersion: Int)
   extends Transformer with HasInputCol with HasOutputCol with HasNumFeatures
     with DefaultParamsWritable {
 
-  private var hashFunc: Any => Int = FeatureHasher.murmur3Hash
+  lazy val hashFunc = hashFuncVersion match {
```

Comment:
This is only used one place; it seems like we could inline this too. I don't know why it was a var to begin with. It would need another if statement in hash(), sure, but that seems trivial.
At least I don't think you need the ` _` in calls below.